openmmtools/mcmc.py

```diff
@@ -656,24 +662,37 @@ def apply(self, thermodynamic_state, sampler_state):
 
         # Perform the integration.
         for attempt_counter in range(self.n_restart_attempts + 1):
-            sampler_state.apply_to_context(context)
 
+            # If we reassign velocities, we can ignore the ones in sampler_state.
+            sampler_state.apply_to_context(context, ignore_velocities=self.reassign_velocities)
```

Comment:
This is just an optimization thing. It should save some time passing information to the GPU at each iteration.